tests/e2e/support/api/share/share.ts

```diff
@@ -32,37 +35,146 @@ export const shareRoles: Readonly<{
   'Secret File Drop': 'uploader'
 } as const
 
-export const createShare = async ({
+const getPermissionsRoleIdByName = (permissionsRole: string): string => {
+  switch (permissionsRole) {
+    case 'viewer':
+      return 'b1e2218d-eef8-4d4c-b82d-0f1a1b48f3b5'
+    case 'space viewer':
+      return 'a8d5fe5e-96e3-418d-825b-534dbdf22b99'
+    case 'editor':
+      return 'fb6c3e19-e378-47e5-b277-9732f9de6e21'
+    case 'space editor':
+      return '58c63c02-1d89-4572-916a-870abc5a1b7d'
+    case 'file editor':
+      return '2d00ce52-1fc2-4dbc-8b95-a73b73395f5a'
+    case 'co owner':
+      return '3a4ba8e9-6a0d-4235-9140-0e7a34007abe'
+    case 'uploader':
+      return '1c996275-f1c9-4e71-abdf-a42f6495e960'
+    case 'manager':
+      return '312c0871-5ef7-4b3a-85b6-0e4074c64049'
+    case 'secure viewer':
+      return 'aa97fe03-7980-45ac-9e50-b325749fd7e6'
+    default:
+      throw new Error(`Role ${permissionsRole} not found`)
+  }
+}
+
+const getRoleId = (role: string, resourceType: string): string => {
+  let roleId: string
+  const shareRole: string = shareRoles[role as keyof typeof shareRoles]
+  if (shareRole === 'editor' && resourceType === 'file') {
+    roleId = getPermissionsRoleIdByName('file editor')
+  } else if (shareRole === 'editor' && resourceType === 'space') {
+    roleId = getPermissionsRoleIdByName('space editor')
+  } else if (shareRole === 'viewer' && resourceType === 'space') {
+    roleId = getPermissionsRoleIdByName('space viewer')
+  } else if (shareRole === 'viewer' && resourceType === 'secure') {
+    roleId = getPermissionsRoleIdByName('space viewer')
+  } else {
+    roleId = getPermissionsRoleIdByName(shareRole)
+  }
+  return roleId
+}
+
+const getObjectId = (shareType: string, shareWith: string): string => {
```

Comment:
what does `object` represent here?

Reply:
it refers to user/recipient id

Reply:
hmm, then the function can be something like `getRecipientId`